File: shared/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/elements/ModuleElement.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client.clickgui.elements;

import net.ccbluex.liquidbounce.LiquidBounce;
import net.ccbluex.liquidbounce.features.module.Module;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import org.lwjgl.input.Mouse;

@SideOnly(Side.CLIENT)
public class ModuleElement extends ButtonElement {

	private final Module module;

	private boolean showSettings;
	private float settingsWidth;
	private boolean wasPressed;

	public int slowlySettingsYPos;
	public int slowlyFade;

	public ModuleElement(final Module module) {
		super(null);

		displayName = module.getName();
		this.module = module;
	}

	@Override
	public void drawScreen(final int mouseX, final int mouseY, final float button) {
		LiquidBounce.clickGui.style.drawModuleElement(mouseX, mouseY, this);
	}

	@Override
	public void mouseClicked(final int mouseX, final int mouseY, final int mouseButton) {
		if (mouseButton == 0 && isHovering(mouseX, mouseY) && isVisible()) {
			module.toggle();
			mc.getSoundHandler().playSound("gui.button.press", 1.0F);
		}

		if (mouseButton == 1 && isHovering(mouseX, mouseY) && isVisible()) {
			showSettings = !showSettings;
			mc.getSoundHandler().playSound("gui.button.press", 1.0F);
		}
	}

	public Module getModule() {
		return module;
	}

	public boolean isShowSettings() {
		return showSettings;
	}

	public void setShowSettings(final boolean showSettings) {
		this.showSettings = showSettings;
	}

	public boolean isntPressed() {
		return !wasPressed;
	}

	public void updatePressed() {
		wasPressed = Mouse.isButtonDown(0);
	}

	public float getSettingsWidth() {
		return settingsWidth;
	}

	public void setSettingsWidth(final float settingsWidth) {
		this.settingsWidth = settingsWidth;
	}
}

File: shared/main/java/net/ccbluex/liquidbounce/ui/client/clickgui/elements/ModuleElement.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.ui.client.clickgui.elements

import net.ccbluex.liquidbounce.LiquidBounce
import net.ccbluex.liquidbounce.features.module.Module
import org.lwjgl.input.Mouse

class ModuleElement(val module: Module) : ButtonElement(module.name)
{
    var showSettings = false
    var settingsWidth = 0f
    private var wasLeftClickPressed = false
    private var wasRightClickPressed = false
    var slowlySettingsYPos = 0
    var slowlyFade = 0

    val isntLeftPressed: Boolean
        get() = !wasLeftClickPressed
    val isntRightPressed: Boolean
        get() = !wasRightClickPressed

    override fun drawScreen(mouseX: Int, mouseY: Int, button: Float)
    {
        LiquidBounce.clickGui.style.drawModuleElement(mouseX, mouseY, this)
    }

    override fun mouseClicked(mouseX: Int, mouseY: Int, mouseButton: Int)
    {
        if (isHovering(mouseX, mouseY) && isVisible && mouseButton in 0..1)
        {
            if (mouseButton == 0) module.toggle() else showSettings = !showSettings

            mc.soundHandler.playSound("gui.button.press", 1.0f)
        }
    }

    fun updatePressed()
    {
        wasLeftClickPressed = Mouse.isButtonDown(0)
        wasRightClickPressed = Mouse.isButtonDown(1)
    }
}
